dpgen/generator/lib/abacus_scf.py

```diff
@@ -22,68 +22,97 @@ def make_abacus_scf_input(fp_params):
-            ret += "force %d\n" % fp_params["force"]
-        elif key == "stress":
-            assert(fp_params["stress"] == 0  or fp_params["stress"] == 1), "'stress' should be either 0 or 1."
-            ret += "stress %d\n" % fp_params["stress"]    
+        elif key == "smearing_method":
+            assert(fp_params["smearing_method"] in ["gaussian", "fd", "fixed", "mp", "mp2", "mv"]), "'smearing_method' should in 'gaussian', 'fd', 'fixed', 'mp', 'mp2', 'mv'. "
+            ret += "smearing_method %s\n" % fp_params["smearing_method"]
+        elif key == "smearing_sigma":
+            if type(fp_params["smearing_sigma"])==str:
+                fp_params["smearing_sigma"] = float(fp_params["smearing_sigma"])
+            assert(fp_params["smearing_sigma"] >= 0), "'smearing_sigma' should be non-negative."
+            ret += "smearing_sigma %f\n" % fp_params["smearing_sigma"]
+        elif key == "cal_force":
+            if type(fp_params["cal_force"])==str:
+                fp_params["cal_force"] = int(eval(fp_params["cal_force"]))
+            assert(fp_params["cal_force"] == 0  or fp_params["cal_force"] == 1), "'cal_force' should be either 0 or 1."
+            ret += "cal_force %d\n" % fp_params["cal_force"]
+        elif key == "cal_stress":
+            if type(fp_params["cal_stress"])==str:
+                fp_params["cal_stress"] = int(eval(fp_params["cal_stress"]))
+            assert(fp_params["cal_stress"] == 0  or fp_params["cal_stress"] == 1), "'cal_stress' should be either 0 or 1."
+            ret += "cal_stress %d\n" % fp_params["cal_stress"]
         #paras for deepks
         elif key == "deepks_out_labels":
+            if type(fp_params["deepks_out_labels"])==str:
+                fp_params["deepks_out_labels"] = int(eval(fp_params["deepks_out_labels"]))
             assert(fp_params["deepks_out_labels"] == 0 or fp_params["deepks_out_labels"] == 1), "'deepks_out_labels' should be either 0 or 1."
             ret += "deepks_out_labels %d\n" % fp_params["deepks_out_labels"]
         elif key == "deepks_descriptor_lmax":
+            if type(fp_params["deepks_descriptor_lmax"])==str:
+                fp_params["deepks_descriptor_lmax"] = int(fp_params["deepks_descriptor_lmax"])
             assert(fp_params["deepks_descriptor_lmax"] >= 0),  "'deepks_descriptor_lmax' should be  a positive integer."
             ret += "deepks_descriptor_lmax %d\n" % fp_params["deepks_descriptor_lmax"]
         elif key == "deepks_scf":
+            if type(fp_params["deepks_scf"])==str:
+                fp_params["deepks_scf"] = int(eval(fp_params["deepks_scf"]))
             assert(fp_params["deepks_scf"] == 0  or fp_params["deepks_scf"] == 1), "'deepks_scf' should be either 0 or 1."
             ret += "deepks_scf %d\n" % fp_params["deepks_scf"]
         elif key == "deepks_model":
             ret += "deepks_model %s\n" % fp_params["deepks_model"]
-        elif key != "k_points": # "k_points key is used to generate KPT file."
```

Comment:
This `elif` is necessary. The `k_points` is also specified in `fp_params`, but can not be written in `INPUT` file.

Reply:
This `k_points` tag as been upgraded on the same level as 'user_fp_param'. Users may also use 'fp_kpt_file' to read KPT file now.

Reply:
Got that. I have no other questions.